### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by mullenlucas**
## Implemented features

- Created a class `Classroom` with the following:
  - `@label` instance variable
  - Setter and getter for `@label`
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`
  - Created the `has-many` side
  - Created the `belongs-to` side
- Created a class `Book`
  - with `@title` and `@author` instance variables,
  - Setters and getters for instance variables
- Created a class `Rental` with the following:
  - `@date` instance variable
  - Setter and getter for `@date`
- Created the `many-to-many` relationship between `Person` and `Book` using the intermediate class `Rental`
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: entities/person.rb [hunk 1]
```diff
@@ -24,6 +25,10 @@ def correct_name
     @name
   end
 
+  def create_rental(_rental)
+    Rental.new(date, self, person)
+  end
+
```
Comment (zieeco) [inline entities/person.rb]: - Let's update the method as we only need the date and book that would rented out by the person below
- Change the person to book as the person is renting a book
``` 
  def create_rental(book, date)
    Rental.new(date, self, book)
  end
```

File: entities/person.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 require_relative '../operations/trim_dc'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (zieeco) [inline entities/person.rb]: - Let's add the :books here also

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @mullenlucas 


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice with the association. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: entities/person.rb (hunk 1 above)
Comment (mullenlucas) [reply entities/person.rb]: Thank you very much for your suggestions !

**Review by victorokoroji — APPROVED** [review]
Hi @mullenlucas, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**